octobot_trading/exchanges/implementations/future_ccxt_exchange.py

```diff
@@ -167,3 +168,66 @@ def clean_trade(self, trade):
 
     def clean_order(self, order):
         return self.connector.clean_order(order)
+
+    async def get_open_positions(self, **kwargs: dict) -> list:
+        return await self.connector.get_open_positions(**kwargs)
+
+    async def get_funding_rate(self, symbol: str, **kwargs: dict) -> dict:
+        return await self.connector.get_funding_rate(symbol=symbol, **kwargs)
+
+    async def get_funding_rate_history(self, symbol: str, limit: int = 1, **kwargs: dict) -> list:
+        return await self.connector.get_funding_rate_history(symbol=symbol, limit=limit, **kwargs)
+
+    async def set_symbol_leverage(self, symbol: str, leverage: int):
+        return await self.connector.set_symbol_leverage(leverage=leverage, symbol=symbol)
+
+    async def set_symbol_margin_type(self, symbol: str, isolated: bool):
+        return await self.connector.set_symbol_margin_type(symbol=symbol, isolated=isolated)
+
+    def parse_position(self, position_dict) -> dict:
+        try:
+            position_dict.update({
+                trading_enums.ExchangeConstantsPositionColumns.SYMBOL.value:
+                    position_dict.get(position_dict[trading_enums.ExchangePositionCCXTColumns.SYMBOL.value]),
+                trading_enums.ExchangeConstantsPositionColumns.ID.value:
+                    position_dict.get(position_dict[trading_enums.ExchangePositionCCXTColumns.SYMBOL.value]),
+                trading_enums.ExchangeConstantsPositionColumns.TIMESTAMP.value:
+                    position_dict.get(trading_enums.ExchangePositionCCXTColumns.TIMESTAMP.value,
+                                      self.connector.get_exchange_current_time()),
+                trading_enums.ExchangeConstantsPositionColumns.SIDE.value:
+                    self.parse_position_side(
+                        position_dict.get(trading_enums.ExchangePositionCCXTColumns.SIDE.value,
+                                          trading_enums.PositionSide.UNKNOWN.value)),
+                trading_enums.ExchangeConstantsPositionColumns.MARGIN_TYPE.value:
+                    position_dict.get(trading_enums.ExchangePositionCCXTColumns.MARGIN_TYPE.value, None),
+                trading_enums.ExchangeConstantsPositionColumns.QUANTITY.value:
+                    decimal.Decimal(
+                        position_dict.get(trading_enums.ExchangePositionCCXTColumns.CONTRACT_SIZE.value, 0)),
+                trading_enums.ExchangeConstantsPositionColumns.COLLATERAL.value:
+                    decimal.Decimal(
+                        position_dict.get(trading_enums.ExchangePositionCCXTColumns.COLLATERAL.value, 0)),
+                trading_enums.ExchangeConstantsPositionColumns.NOTIONAL.value:
+                    decimal.Decimal(
+                        position_dict.get(trading_enums.ExchangePositionCCXTColumns.NOTIONAL.value, 0)),
+                trading_enums.ExchangeConstantsPositionColumns.LEVERAGE.value:
+                    decimal.Decimal(
```

Comment:
We might need to do `f"{position_dict.get(trading_enums.ExchangePositionCCXTColumns.LEVERAGE.value, 0)}"` to ensure no rounding issues (creating a decimal from float can add .99999999 etc numbers, it doesn't happen from str)

Reply:
Right :+1: